README.md

```diff
@@ -1,73 +1,295 @@
+their topic/service types and the data exchange flow. This configuration file is loaded during
+runtime, so there is no need to recompile any package before switching to a whole new
+intercommunication architecture.
+
+To get a more precise idea on how these YAML files have to be filled and which fields they require
+in order to succesfully configure and launch an *Integration Service* project, please refer to the
+[dedicated configuration section](https://integration-service.docs.eprosima.com/en/latest/yaml_config.html) of the official documentation.
+
+Regarding the *WebSocket System Handle*, there are several specific parameters which can be configured
+for the WebSocket middleware. All of these parameters fall as suboptions of the main
+five sections described in the *Configuration* chapter of *Integration Service* repository:
+
+* `systems`: The system `type` must be `websocket_server` or `websocket_client`. In addition to the
+  `type` and `types-from` fields, the *WebSocket System Handle* accepts a wide variety of specific
+  configuration fields, depending on the selected operation mode (*Client* or *Server*):
+
+  For the `websocket_server` *System Handle*, there are two possible configuration scenarios:
+  the former one uses a TLS endpoint, and the latter uses a TCP endpoint.
+
+  **TLS**
+  ```yaml
+  systems:
+    websocket:
+      type: websocket_server
+      port: 80
+      cert: path/to/cert/file.crt
+      key: path/to/key/file.key
+      authentication:
+       policies: [
+           { secret: this-is-a-secret, algo: HS256, rules: {example: *regex*} }
+       ]
+  ```
+
+  **TCP**
+  ```yaml
+  systems:
+    websocket:
+      type: websocket_server
+      port: 80
+      security: none
+      encoding: json
+      authentication:
+       policies: [
+           { secret: this-is-a-secret, algo: HS256, rules: {example: *regex*} }
+       ]
+  ```
+
+    * `port`: The specific port where the *server* will listen for incoming connections. This field is
+      required.
+    * `security`: If this field is not present, a secure TLS endpoint will be created. If the special
+      value `none` is written, a TCP *WebSocket server* will be set up.
+    * `cert`: The *X.509* certificate that the *server* should use. This field is mandatory if
+      `security` is enabled.
+    * `key`: A path to the file containing the public key used to verify credentials with the specified
+      certificate. If `security` is enabled, this field must exist and must be filled in properly.
+    * `authentication`: It is a list of `policies`. Each policy accepts the following keys:
+      * `secret`: When using **MAC** *(Message Authentication Code)* method for verification,
+        this field allows to set the secret used to authenticate the client requesting a connection to the server.
+      * `pubkey`: Path to a file containing a **PEM** encoded public key.
+
+      > **_NOTE:_** Either a `secret` or a `pubkey` is required.
```

Comment:
I think it would be better if it is indented to the same level as `pubkey` and `secret`